### PR TITLE
Renamed interface and split into two classes

### PR DESCRIPTION
**Pull request by tuhrig**

### CODE REVIEW

File: src/test/kotlin/de/bringmeister/spring/aws/kinesis/AwsKinesisClientProviderTest.kt [hunk 1]
```diff
@@ -20,17 +20,17 @@ class AwsKinesisClientProviderTest {
     }
 
     val credentialsProvider: AWSCredentialsProvider = mock { }
-    val assumeRoleCredentialsProviderFactory: AssumeRoleCredentialsProviderFactory = mock { }
+    val AWSCredentialsProviderFactory: AWSCredentialsProviderFactory = mock { }
```
Comment (jmatusewicz-bringmeister) [inline src/test/kotlin/de/bringmeister/spring/aws/kinesis/AwsKinesisClientProviderTest.kt]: here as well: variable shouldn't start with capital letter

File: src/main/kotlin/de/bringmeister/spring/aws/kinesis/AWSCredentialsProviderFactory.kt [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+import com.amazonaws.auth.AWSCredentialsProvider
+
+/**
+ * Every consumer/listener will have its own credentials. Those credentials
+ * will include the IAM role of the consumer/listener which he needs in order
+ * to read from a stream. This factory will create those credentials for a given
+ * role. The role to assume is configured in the application properties for each
+ * consumer.
+ *
+ * Note: We need this interface because have two implementations of this class.
```
[the remark this answers is not in the document]
Comment (tuhrig) [reply src/main/kotlin/de/bringmeister/spring/aws/kinesis/AWSCredentialsProviderFactory.kt]: you read the comments? wow :)

File: src/main/kotlin/de/bringmeister/spring/aws/kinesis/AwsKinesisAutoConfiguration.kt [hunk 3]
```diff
@@ -19,10 +19,10 @@ class AwsKinesisAutoConfiguration {
     @ConditionalOnMissingBean
     fun kinesisClientProvider(consumerClientConfigFactory: ConsumerClientConfigFactory,
                               producerClientFactory: ProducerClientFactory,
-                              kinesisCredentialsProviderFactory: AssumeRoleCredentialsProviderFactory,
+                              AWSCredentialsProviderFactory: AWSCredentialsProviderFactory,
```
[the remark this answers is not in the document]
Comment (tuhrig) [reply src/main/kotlin/de/bringmeister/spring/aws/kinesis/AwsKinesisAutoConfiguration.kt]: Sure

File: src/test/kotlin/de/bringmeister/spring/aws/kinesis/AwsKinesisClientProviderTest.kt (hunk 1 above)
Comment (tuhrig) [reply src/test/kotlin/de/bringmeister/spring/aws/kinesis/AwsKinesisClientProviderTest.kt]: Sure

**Comment by jmatusewicz-bringmeister** [comment]
👍

**Pull request merged**